Makefile

```diff
@@ -25,25 +19,44 @@ SOURCEC = $(SRC)/main.cpp \
 	$(SRC)/veloc_total.cpp \
 	$(SRC)/sp.cpp
 OBJECTS = $(SOURCEC:%.cpp=%.o)
+BIN_DIR = bin
+MANDYOC = $(BIN_DIR)/mandyoc
+LOCAL_BIN = $(HOME)/.local/bin
+
+.PHONY: help all insatall clear test
 
 help:
 	@echo ""
 	@echo "Commands:"
 	@echo ""
-	@echo "  all		Build and install Mandyoc by running"
-	@echo "  test_mandyoc	Run the Mandyoc test using 2 cores. It takes several munutes"
+	@echo "  all		Build Mandyoc"
+	@echo "  install	Install MANDYOC in ~/.local/bin/"
+	@echo "  test		Run the MANDYOC test using 1 cores. It takes several munutes"
+	@echo "  clear		Removes the files produced when building MANDYOC"
+	@echo ""
+
+all: $(MANDYOC)
+	@echo ""
+	@echo "Mandyoc built in bin/"
+
+install: $(MANDYOC)
 	@echo ""
+	install $< $(LOCAL_BIN)
 
-# Run test
-test_mandyoc:
-	@echo "Run MANDYOC test may take several minutes.."
-	cd test/testing_data/ ; ${MPIEXEC} -n 2 ../../mandyoc
+test:
+	@echo "Run MANDYOC test may take several minutes..."
+	cd test/testing_data/ ; mpirun -n 1 mandyoc
 	pytest -v test/testing_result.py
 
-# Build Mandyoc
-all: ${OBJECTS} chkopts
-	-${CLINKER} -o mandyoc ${OBJECTS} ${PETSC_LIB}
+clear:
 	rm $(SRC)/*.o
+	rm -rf $(BIN_DIR)
```

Comment:
I think this is too dangerous. Any miss configuration in these paths (or any mistake) can remove not intended user data.

Reply:
This only work in the repository folder. `bin` is a folder created during mandyoc build inside the repository folder. 
`crear` doesn't remove your bin folder

Reply:
Yes, if used correctly there is no problem. But, I'm confused

```
BIN_DIR = bin
MANDYOC = $(BIN_DIR)/mandyoc
LOCAL_BIN = $(HOME)/.local/bin
```

LOCAL_BIN is referencing **bin**, but should it be the same bin in BIN_DIR?
In this case we are left with $(HOME)/.local/bin.

For instance, what you have if user accidentally left BIN_DIR empty?

I think the best approach is try to use the local dir, the folder in which mandyoc was cloned to, so we don't reduce a lot a case where *accidentally* we (force) delete something involving $(HOME)/.local.

Reply:
@rafaelmds, I'm not sure what your doubt is. But I try to answer you...

> LOCAL_BIN is referencing bin, but should it be the same bin in BIN_DIR?
> In this case we are left with $(HOME)/.local/bin.
> instance, what you have if user accidentally left BIN_DIR empty?
They are different, for example: The BIN_DIR path is ~/madyoc/bin and the LOCAL_BIN path is ~/.local/bin.
I create this bin folder in the repository to keep everything more organized. So the mandyoc binary will be in this folder and not in the root path. 
When you run `all`, It builds the mandyoc binary into the BIN_DIR. When you run `install`, it installs mandyoc binary in the user system in the LOCAL_BIN.
So if you debate this folder BIN_DIR, you lost the mandyoc binary, but you still run mandyoc because it is installed in your system. 

> I think the best approach is try to use the local dir, the folder in which mandyoc was cloned to, so we don't reduce a lot a case where accidentally we (force) delete something involving $(HOME)/.local.
If you use `cliner` you don't touch your system folder, it only  deletes the ~/mandyoc/bin

Reply:
thanks for this explanation @aguspesce :smile:

But I still prefer hardcode this bin directory, and use 
```
MANDYOC = bin/mandyoc
```
Also, we could keep the bin directory in the repo and ignore everything inside it. In this case, a tweak in $(BIN_DIR) action would be be necessary:
```
$(BIN_DIR):
	mkdir -p bin
```

To keep empty bin dir, a gitignore file (`bin/.gitignore`) can be used, with contents:
```
*
!.gitignore
```
Then, the clear action would be like this (to keep bin folder)
```
clear:
	rm $(SRC)/*.
        ls bin/* | grep -v '(.gitignore)' | xargs rm
```

Let me know what you think...